lsp/nls/src/linearization.rs

```diff
@@ -74,6 +93,22 @@ pub struct AnalysisHost {
     env: Environment,
     scope: Vec<ScopeId>,
     meta: Option<MetaValue>,
+    /// Indexing a record will store a reference to the record as
+    /// well as its fields.
+    /// [Self::Scope] will produce a host with a single **`pop`ed**
+    /// Ident. As fields are typechecked in the same order, each
```

Comment:
The same hashmap object does not change the order indeed, thats what I am relying on here. The same object is iterated twice, once by the linearizer and once by the typechecker. I do need to somehow match definitions (which immediately follow the record lin.-item) with their value.
Technically the order is not guaranteed by rust to _never_ change, we should keep that in mind or add a comment about that.

Reply:
Then, maybe you want to sort the items before iteration, as done in serialization to guarantee order ( `serialize::serialize_record`) ? Otherwise if that ever changes, I imagine this would be a difficult thing to debug.

Reply:
This would mean cloning every single record without a change in behaviour (except deterministic order). I am not sure if this is a reasonable performance implication

Reply:
Let's leave this to be decided with #471 which actually uses the fields

Reply:
Well, you mean cloning the keys, which I think wouldn't that bad. Because we wouldn't be cloning the record's actual content, obviously. But that's a fair remark. Ok to postpone this to #471.